.github/workflows/easy-cp.yml

```diff
@@ -0,0 +1,81 @@
+# Copyright 2023 The Flutter Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+name: Cherry-pick Labeled PR to Release Branch
+
+on:
+  pull_request:
+    branches: master
+    types: [labeled]
+
+permissions: write-all
+
+jobs:
+  cherrypick_to_release:
+    name: cherrypick_to_release
+    runs-on: ubuntu-latest
+    if: |
+      (github.event.label.name == format('cp{0} beta', ':') || github.event.label.name == format('cp{0} stable', ':')) &&
+      (github.event.pull_request.merged == true)
+    steps:
+      - name: Get Release Channel
+        run: |
+          echo "CHANNEL=$(echo ${{ github.event.label.name }} | cut -d ':' -f 2 | xargs)" >> $GITHUB_ENV
+      - name: Get Release Candidate Branch
+        run: |
+          curl -o version.json https://raw.githubusercontent.com/flutter/flutter/$CHANNEL/bin/internal/release-candidate-branch.version
+          echo "RELEASE_BRANCH=$(cat version.json | tr -d '\n')" >> $GITHUB_ENV
+          rm version.json
+      - name: Get Cherry Pick PR
+        run: |
+          echo "COMMIT_SHA=$(echo ${{ github.event.pull_request.merge_commit_sha }})" >> $GITHUB_ENV
+      - name: Checkout Flutter Repo
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11
+        with:
+          repository: flutter/flutter
+          token: ${{ github.token }}
+          path: flutter
+          ref: master
+          fetch-depth: 0
+      # use same name when checking out branch, since the marketplace action does a hard reset.
+      - name: Attempt CP
+        id: attempt-cp
+        working-directory: ./flutter
+        run: |
+          git config user.name "GitHub Actions Bot"
+          git config user.email "<>"
+          git fetch origin $RELEASE_BRANCH
+          git fetch origin master
+          git checkout -b cp-$COMMIT_SHA --track origin/$RELEASE_BRANCH
+          git cherry-pick $COMMIT_SHA
+          git status
+      # TODO(xilaizhang): move cp template to flutter/flutter.
+      # The Pull Request cherry pick template should be under flutter/.github/PULL_REQUEST_CP_TEMPLATE.md.
+      - name: Get CP Template
+        run: |
+          curl -o flutter/.github/PULL_REQUEST_CP_TEMPLATE.md https://raw.githubusercontent.com/XilaiZhang/miscellaneous-side-project/master/.github/workflows/template/cherrypick.md
+      # PAT token is needed, otherwise 'Permission to flutter/flutter.git denied to github-actions[bot].'
+      - name: Create PR on CP success
+        if: ${{ steps.attempt-cp.conclusion == 'success' }}
+        uses: peter-evans/create-pull-request@153407881ec5c347639a548ade7d8ad1d6740e38
+        with:
+          base: ${{ env.RELEASE_BRANCH }}
+          path: flutter
+          token: ${{ secrets.PAT }}
+          commit-message: automated cherry pick
```

Comment:
Can we re-use the original commit message?

Reply:
My understanding is that there isn't an original commit message, because the `message` associated with a merged commit sha (cp commit sha), is the PR description of the merged PR.

e.g. the merged commit sha is af61c6c73fb1057633c84bde5b183f8fc3b0673a of https://github.com/flutter/flutter/pull/139348. running the query `gh search commits --repo flutter/flutter --hash e5762ba --json commit` gives a commit message that is several hundred chars (the pr description)

e.g. in the cherry pick pr https://github.com/flutter/flutter/pull/139594. The commit message associated with the cherry picked commit is the original PR title.

Reply:
Do we need the create pull request action? Can we use the gh cli directly?

Reply:
Yeah we can definitely use gh cli directly. 

My first few drafts are all implemented with pure gh/git commands, but I think in previous design/code reviews it was pointed out that I should use market place actions whenever possible. I am good with reimplementing with gh and git commands again. 

However, if the concern here is that an unnecessary commit message will be generated, then I can assure that no duplicate/extra commit message will be produced. See https://github.com/flutter/flutter/pull/139843 as an example where I cp-ed a chillers pr using market place action.